lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb

```diff
@@ -93,37 +93,46 @@ def self.association=(meth)
 
     def self.base_class
       @base_class ||= begin
-        model_name_from_active_record_model(model.base_class).constantize
+        find_or_create_service_model(model.base_class)
       end
     end
 
     def self.base_model
       @base_model ||= begin
-        model_name_from_active_record_model(model.base_model).constantize
+        find_or_create_service_model(model.base_model)
       end
     end
 
     def self.model
       # Set a class-instance variable to get the appropriate model
-      @model ||= /MiqAeService(.+)$/.match(name)[1].gsub(/_/, '::').constantize
+      @model ||= service_model_name_to_model(name)
```

Comment:
good catch. this will cache the old model.

I am not really seeing these used too much, so not sure exactly why they are cached. e.g.: I don't see `ar_model` used anywhere

The associations are cached, but I think those may be ok.